File: app/src/main/java/com/pysiak/simple/clock/ui/SimpleClockActivity.java
package com.pysiak.simple.clock.ui;

import android.os.Bundle;
import androidx.annotation.Nullable;
import com.google.android.material.tabs.TabLayout;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.Toast;
import com.pysiak.simple.clock.R;
import java.util.List;

/**
 * Main class, launcher activity for application, implementation of {@link SimpleClockView}
 *
 * @author Yaroslav P.
 */
public class SimpleClockActivity extends AppCompatActivity implements SimpleClockView {

    private ViewPager viewPager;
    private TabLayout tabLayout;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tabLayout = findViewById(R.id.tab_layout_id);
        viewPager = findViewById(R.id.view_pager_id);
        tabLayout.setupWithViewPager(viewPager);
        SimpleClockPresenter simpleclockPresenter = new SimpleClockPresenter(this);
        simpleclockPresenter.attachView(this);
        simpleclockPresenter.loadData();
    }

    @Override
    public void setPagerAdapter(List<Fragment> fragmentList, List<String> titleList, List<Integer> iconList) {
        if (fragmentList != null && titleList != null && fragmentList.size() == titleList.size()) {
            TabAdapter adapter = new TabAdapter(getSupportFragmentManager());
            for (int i = 0; i < fragmentList.size(); i++) {
                adapter.addFragment(fragmentList.get(i), titleList.get(i));
            }
            viewPager.setAdapter(adapter);
            if (iconList != null && tabLayout.getTabCount() == iconList.size()) {
                for (int j = 0; j < iconList.size(); j++){
                    tabLayout.getTabAt(j).setIcon(iconList.get(j));
                }
            }
        } else {
            Toast toast = Toast.makeText(getApplicationContext(), "Check your list size", Toast.LENGTH_SHORT);
            toast.show();
        }
    }
}

File: app/src/main/java/com/pysiak/simple/clock/ui/SimpleClockActivity.kt
package com.pysiak.simple.clock.ui

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.view.ViewPager
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import com.pysiak.simple.clock.R
import com.pysiak.simple.clock.presenter.ClockPresenter
import com.pysiak.simple.clock.presenter.TabAdapter
import com.pysiak.simple.clock.presenter.SimpleClockView


class SimpleClockActivity : AppCompatActivity(), SimpleClockView {

   private lateinit var viewPager: ViewPager
   private lateinit var tabLayout: TabLayout
   private lateinit var presenter: ClockPresenter



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        viewPager = findViewById(R.id.viewPagerId)
        tabLayout = findViewById(R.id.tabLayoutId)

        tabLayout.setupWithViewPager(viewPager)
        presenter = ClockPresenter(this)
        presenter.attachView(this)
        presenter.loadData()


    }
    override fun setPagerAdapter(fragmentList: List<Fragment>,titleList: List<String>) {
        val adapter = TabAdapter(supportFragmentManager)

        for (i in 0..fragmentList.size-1) {
            adapter.addFragment(fragmentList[i], titleList[i])
        }
        viewPager.adapter = adapter

        
    }
}























/*adapter = TabAdapter(supportFragmentManager)
           adapter.addFragment(TabFragment().newInstance("Tab1"), "Tab 1")
           adapter.addFragment(TabFragment().newInstance("Tab2"), "Tab 2")
           adapter.addFragment(TabFragment().newInstance("Tab3"), "Tab 3")
   viewPager.adapter = adapter
       tabLayout.setupWithViewPager(viewPager)*/